### DISCUSSION TITLE
Refreshing Virtual Datasets

### DISCUSSION THREAD

**Discussion by maddyobrienjones** [apache/superset#13325] [Q&A / Help] [answered]
Most of my charts in Superset are based off of queries on datasets that are continuously uploaded which I have turned into virtual datasets. Do virtual datasets refresh automatically when the source datasets are uploaded, or do they need to be refreshed manually? My datasets update quite frequently, and I don't want to have to be going in and refreshing a bunch of queries each time.

**Comment by maddyobrienjones** [comment, marked as answer]
Just tested and they do update automatically.

**Reply by villebro** [reply]
Yes, virtual datasets are just a drop-in replacement for the table name in the chart query, so the dataset isn't materialized in Superset.

Regular physical table (the table is called `my_table`):
```sql
SELECT ...
FROM my_table
WHERE ...
GROUP BY ...
```

Virtual table (here `SELECT x FROM y` is the virtual dataset query):
```sql
SELECT ...
FROM (SELECT x FROM y) AS expr_qry
WHERE ...
GROUP BY ...
```
